### PR TITLE
assessment week 3 complete

### PR DESCRIPTION
**Pull request by jumba23**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ x] Coding Challenges
- [ x] Interview Questions
- [ x] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -10,18 +18,68 @@
...
+  });
+});
+
+//ReferenceError: fibSequenceNum is not defined
+
 // b) Create the function that makes the test pass.
 
+// ============== <<<<< PSEUDO CODE >>>>> ===========================
+// the function will take in a number that will essentially set the fibonacci array's length
+//we need to create fibonacci array using simple formula or we can just copy an array from internet,
```
Comment (elysemontano) [inline code-challenges.test.js:55]: Not entirely sure what you are meaning here, but would strongly recommend not copying from the internet.

File: code-challenges.test.js [hunk 2]
```diff
@@ -10,18 +18,68 @@
...
+describe("fibSequenceNum", () => {
+  it("The function takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence", () => {
+    const fibLength1 = 6;
+    const fibLength2 = 10;
+    const fibLength3 = 1;
+    expect(fibSequenceNum(fibLength1)).toEqual([1, 1, 2, 3, 5, 8]);
+    expect(fibSequenceNum(fibLength2)).toEqual([
+      1, 1, 2, 3, 5, 8, 13, 21, 34, 55,
+    ]);
+    expect(fibSequenceNum(fibLength3)).toEqual("1 in NOT greater than 2");
```
Comment (elysemontano) [inline code-challenges.test.js:45]: Nice work on testing for an edge case!

File: code-challenges.test.js [hunk 3]
```diff
@@ -45,23 +103,105 @@ const studyMinutesWeek2 = {
...
+//since this is not an array, we  need to use Object.values function to access the values only
+//we are going to chain .forEach function so we can iterate the obj elements. Each value/element will be pushed into our newArray, one at a time.
+//the function will return sorted version of our new array
+
+const ascendingArray = (obj) => {
+  const newArray = [];
+  Object.values(obj).forEach((val) => {
+    newArray.push(val);
+  });
+  return newArray.sort((a, b) => a - b);
```
Comment (elysemontano) [inline code-challenges.test.js:151]: You can chain .sort to the array given from Object.values removing the need for .forEach.

File: code-challenges.test.js [hunk 4]
```diff
@@ -45,23 +103,105 @@ const studyMinutesWeek2 = {
...
+});
+
+//ReferenceError: sumSumArray is not defined
+
+// ============== <<<<< PSEUDO CODE >>>>> ===========================
+// the function will receive an array and return the array of same length. We can use .map to iterate
+// we need to work on logic regarding calculations of our sum for each iteration and push into our new array
+// we need to declare two variables and assign "0" to one of them to help with first .map iteration
+
+const cumSumArray = (arr) => {
```
Comment (elysemontano) [inline code-challenges.test.js:193]: Awesome work!!  Really happy to see you using higher-order functions on this problem! 👏

File: interview-questions.md [hunk 5]
```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: the Document Object Model(DOM) is a programing interface for web documents. The DOM sets the logical structure of documents and the way a document is accessed and manipulated. DOMs are usually implemented to provide scripting access to the content of documents with markup languages such as XML or HTML. In the DOM structure, there are three levels of nodes: The Document-level node, Element nodes, and Text nodes.
 
 6. STRETCH: Which is the difference between a div and a span?
 
-Your answer:
+Your answer: Div is block element used to organize page elements into groups and style them. While span is an inline element used mostly to style some kinds of text.
 
-Researched answer:
+Researched answer: The div should be used to wrap sections of a document, while spans are used to wrap small portions of text, images, etc. The div elements are used to provide base level layout in html, while span element is used to stylize texts.
```
Comment (elysemontano) [inline interview-questions.md:43]: Awesome answers in this section!!  💯

File: react-debugging.html [hunk 6]
```diff
@@ -63,13 +63,18 @@
           padding: "10px"
         }
         // No need to change anything ABOVE this line ^
+         
+        // 1. missing all encompassing return fragment
+        // 2. onClick button does not have any functionality. Missing function declaration 
+        // 3. next button element onclick needs to be camel casing, e.g onClick
```
Comment (elysemontano) [inline react-debugging.html:69]: 🎉

**Review by elysemontano — APPROVED** [review]
Crushed this assessment Goran!  I really like seeing the edge cases in your test and functionality.  Your interview answers were fantastic, great work!
